lib/engine-addon.js

```diff
@@ -387,10 +387,76 @@ module.exports = {
         return mergeTrees([
           externalTree,
           engineJSTree,
-          engineCSSTree
+          compiledEngineCSSTree
         ].filter(Boolean), { overwrite: true });
       };
 
+      this.compileLazyEngineStyles = function compileLazyEngineStyles(vendorTree, externalTree) {
+        var vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
+        var engineStylesOutputDir = 'engines-dist/' + this.name + '/assets/';
+
+        // Get base styles tree.
+        var engineStylesTree = this._treeFor('addon-styles');
```

Comment:
Should we memoize/cache the result of `_treeFor('addon-styles')`? Since it is called in more than one location.